components/cdc/src/channel.rs

```diff
@@ -244,22 +252,57 @@ impl From<MemoryQuotaExceeded> for SendError {
     }
 }
 
+pub struct ObservedEvent {
+    pub created: Instant,
+    pub event: CdcEvent,
+    pub size: usize,
+}
+
+pub struct ScanedEvent {
+    pub created: Instant,
+    pub event: CdcEvent,
+    pub size: usize,
+    pub truncated: Arc<AtomicBool>,
+}
+
+impl ObservedEvent {
+    fn new(created: Instant, event: CdcEvent, size: usize) -> Self {
+        ObservedEvent {
+            created,
+            event,
+            size,
+        }
+    }
+}
+
+impl ScanedEvent {
+    fn new(created: Instant, event: CdcEvent, size: usize, truncated: Arc<AtomicBool>) -> Self {
+        ScanedEvent {
+            created,
+            event,
+            size,
+            truncated,
+        }
+    }
+}
+
 #[derive(Clone)]
 pub struct Sink {
-    unbounded_sender: UnboundedSender<(Instant, CdcEvent, usize)>,
-    bounded_sender: Sender<(Instant, CdcEvent, usize)>,
+    unbounded_sender: UnboundedSender<ObservedEvent>,
+    bounded_sender: Sender<ScanedEvent>,
     memory_quota: Arc<MemoryQuota>,
 }
 
 impl Sink {
+    /// Only observed events can be sent by `unbounded_send`.
     pub fn unbounded_send(&self, event: CdcEvent, force: bool) -> Result<(), SendError> {
```

Comment:
How about change `event` to `ObservedEvent` to make sure that we pass the correct events in. Ditto for `send_all`.